sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -91,6 +96,59 @@ protected JsonSerializable(byte[] bytes) {
         this.propertyBag = fromJson(bytes);
     }
 
+    public static JsonSerializable instantiateFromObjectNodeAndType(ObjectNode objectNode, Class<?> klassType) {
+        if (klassType.equals(Document.class)) {
+            return new Document(objectNode);
+        }
+        if (klassType.equals(InternalObjectNode.class)) {
+            return new InternalObjectNode(objectNode);
+        }
+        if (klassType.equals(PartitionKeyRange.class)) {
+            return new PartitionKeyRange(objectNode);
+        }
+        if (klassType.equals(Range.class)) {
+            return new Range<>(objectNode);
+        }
+        if (klassType.equals(QueryInfo.class)) {
+            return new QueryInfo(objectNode);
+        }
+        if (klassType.equals(PartitionedQueryExecutionInfoInternal.class)) {
+            return new PartitionedQueryExecutionInfoInternal(objectNode);
+        }
+        if (klassType.equals(QueryItem.class)) {
+            return new QueryItem(objectNode);
+        }
+        if (klassType.equals(Address.class)) {
+            return new Address(objectNode);
+        }
+        if (klassType.equals(DatabaseAccount.class)) {
+            return new DatabaseAccount(objectNode);
+        }
+        if (klassType.equals(DatabaseAccountLocation.class)) {
+            return new DatabaseAccountLocation(objectNode);
+        }
+        if (klassType.equals(ReplicationPolicy.class)) {
+            return new ReplicationPolicy(objectNode);
+        }
+        if (klassType.equals(ConsistencyPolicy.class)) {
+            return new ConsistencyPolicy(objectNode);
+        }
+        if (klassType.equals(DocumentCollection.class)) {
+            return new DocumentCollection(objectNode);
+        }
+        if (klassType.equals(Database.class)) {
+            return new Database(objectNode);
+        } else {
+            // This should rarely execute. Keeping this for sanity sake
+            try {
```

Comment:
IMO this should just throw to capture any future regressions early.

Reply:
I will test without this and remove this in a future PR.